rotkehlchen/exchanges/manager.py

```diff
@@ -82,11 +83,20 @@ def get_exchange(self, name: str, location: Location) -> Optional[ExchangeInterf
 
         return None
 
+    def get_excluded_exchanges(self) -> Iterable[ExchangeLocationID]:
+        if not hasattr(self, 'database'):
+            return []
+        db_settings = self.database.get_settings()
+        return db_settings.non_syncing_exchanges
+
     def iterate_exchanges(self) -> Iterator[ExchangeInterface]:
         """Iterate all connected exchanges"""
+        excluded = set(self.get_excluded_exchanges())
```

Comment:
@LefterisJP I've added set() here to make `if exchange.location_id() not in excluded:` work faster (because check of existence in set works faster than in list). But if it's not needed I can surely remove it.